src/common/com/intellij/plugins/haxe/haxelib/HaxelibProjectUpdater.java

```diff
@@ -1495,15 +1491,15 @@ public ProjectTracker getUpdatingProject() {
      */
     private void queueNextProject() {
       synchronized (updateSyncToken) {
-        LOG.assertTrue(null == updatingProject);
+        LOG.assertLog(null == updatingProject, "Assertion failed");
```

Comment:
Seeing how many of these had to be changed, why not add 'assertTrue()' to HaxeDebugLogger?

Reply:
you're right, but I didn't see a big reason to improve/style current `HaxeDebugLogger` as it works only because JetBrains hasn't terminated `log4j` proxy yet: current logger implementation was all about transparent integration with `log4j` and it's about time to make another big decision on how to move to [standard platform logging API](https://plugins.jetbrains.com/docs/intellij/ide-infrastructure.html?_gl=1*1hnudjc*_ga*ODA0NTg5NTYzLjE2Nzg3NTY0NTU.*_ga_9J976DJZ68*MTY4MDI2MzIwMC4zOC4xLjE2ODAyNjMyMzEuMjkuMC4w&_ga=2.158107258.1540209370.1680263200-804589563.1678756455#logging) and revamp `HaxeDebugLogger` from ground up

Reply:
Well, rewriting an API on top of another framework doesn't negate the benefit of a simpler API.  But that's enough on that subject...

If you are interested in putting it on top of another [newer] standard (to which I have no objection), then you should know that HaxeDebugLogger's original purpose was to be able to _suppress_ and redirect IDEA's standard logging during unit tests.  Logging was not only overly-verbose sometimes, but it got in the way of detecting proper functioning of the tested code.  Over time, the class gained more functionality.  If you want to see which tests were involved, you can go back to the commit where the HaxeDebugLogger was originally introduced.

Reply:
Also, if you want to propose that type of change (I see a larger and more useful conversation), let's make a new issue and discuss the change there.